src/address/index.ts

```diff
@@ -75,6 +75,28 @@ export abstract class Address {
     }
   }
 
+  // based on https://github.com/filecoin-project/lotus/blob/80aa6d1d646c9984761c77dcb7cf63be094b9407/chain/types/ethtypes/eth_types.go#L370
+  static fromEthAddress = (network: Network, ethAddr: Buffer): Address => {
+    const idMask = Buffer.alloc(12)
+    idMask[0] = 0xff
+
+    if (idMask == ethAddr.slice(0,12)) {
+      return AddressId.fromBytes(network, ethAddr.slice(12))
+    }
+
+    return new FilEthAddress(network, ethAddr)  
+  }
+
+  static fromEthAddressHex = (network: Network, ethAddr: String): Address => {
```

Comment:
We could combine both fromEthAddressHex and fromEthAddres in one function by combining types fro ethAddr parameter.

Reply:
I would avoid mixing parameters type. It is less clean and makes it more difficult to read.